code/workspaces/web-app/src/components/modal/SiteLogs.spec.js

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import Logs from './SiteLogs';
+import PrimaryActionButton from '../common/buttons/PrimaryActionButton';
+
+function shallowRender(props) {
+  return shallow(<Logs {...props} />);
+}
+
+describe('Logs', () => {
+  const onCancelMock = jest.fn();
+
+  const generateProps = () => ({
+    title: 'Title',
+    body: 'Body',
+    onCancel: onCancelMock,
+  });
+
+  beforeEach(() => jest.resetAllMocks());
+
+  it('creates correct snapshot for logs dialog', () => {
+    // Arrange
+    const props = generateProps();
+
+    // Act
+    const output = shallowRender(props);
+
+    // Assert
+    expect(output).toMatchSnapshot();
+  });
+
+  it('wires up close function correctly', () => {
+    // Arrange
+    const props = generateProps();
+
+    // Act
+    const output = shallowRender(props);
+    const buttons = output.find(PrimaryActionButton);
+    const cancelFunction = buttons.find({ children: 'Close' }).prop('onClick');
+    cancelFunction();
```

Comment:
You can use enzyme to simulate a click rather than finding the function and then calling it in the test.
I think it would look something like
```js
const buttons = output.find(PrimaryActionButton);
const cancelFunction = buttons.find({ children: 'Close' }).simulate('click');
```

I have used it somewhere in the other tests if you are interested and want a working example

Reply:
Ah that's useful to know thanks!